background.js

```diff
@@ -1370,26 +1388,25 @@ function mainLoop() {
       messenger.browserAction.setTitle({title: `${extName} [${isActiveMessage}]`});
 
       let doSequential = preferences.throttleDelay > 0;
-      SLTools.forAllDrafts(SendLater.possiblySendMessage, doSequential).then(() => {
-        SLTools.countActiveScheduledMessages().then(nActive => {
-          SendLater.updateStatusIndicator(nActive);
-          SendLater.setQuitNotificationsEnabled(preferences.askQuit, nActive);
-        });
+      try {
+        await SLTools.forAllDrafts(SendLater.possiblySendMessage, doSequential)
```

Comment:
Since the handling is not awaited, the mainloop will be re-triggered/re-scheduled right after the call has been placed on the stack. Which means the next call could startup, while this one is still handling the same message.